hm/model/host.py

```diff
@@ -109,16 +122,24 @@ def scale(self):
             raise e
 
     @classmethod
-    def _current_group_alternate(cls, group, conf=None):
-        alternates_count = int(config.get_config("HM_ALTERNATIVE_CONFIG_COUNT", 1, conf))
+    def _current_group_alternate(cls, group, conf, alternative_id_error):
+        alternates_count = range(int(config.get_config("HM_ALTERNATIVE_CONFIG_COUNT", 1, conf)))
+        if len(alternative_id_error) == len(alternates_count):
+            raise NoMoreHMAlternativesAvailable
+        alternates_valid = list(set(alternates_count) - set(alternative_id_error))
         hosts = model.storage(conf).list_hosts({'group': group})
         alterantives_map = collections.defaultdict(int)
         for host in hosts:
             alterantives_map[host.alternative_id] += 1
-        min_alt_id = 0
         min_alt_count = None
-        for i in xrange(alternates_count):
+        for i in alternates_valid:
+            if i in alternative_id_error:
```

Comment:
Is this if necessary since you already subtract the alternative_id_error set on line 129?